[hop free symbols][refactor] make bound_symbols a dictionary

ydwu4 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #138363
* #139663
* #138559
* #138737
* #138558
* #138428
* __->__ #138345

Code refactoring only. Change all self.tx.output.bound_symbols to self.tx.output.root_tracer.bound_symbols.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

ydwu4 committed:
[hop][lift free symbols] refactor bound_symbols to be dictionary

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop][lift free symbols] refactor bound_symbols to be dictionary"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop][lift free symbols] refactor bound_symbols to be dictionary"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop][lift free symbols] refactor bound_symbols to be dictionary"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop][lift free symbols] refactor bound_symbols to be dictionary"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop][lift free symbols] refactor bound_symbols to be dictionary"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop][lift free symbols] refactor bound_symbols to be dictionary"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop free symbols][refactor] make bound_symbols a dictionary"


Code refactoring only. Change all self.tx.output.bound_symbols to self.tx.output.root_tracer.bound_symbols.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

zou3519 commented on torch/_dynamo/variables/builder.py:
```diff
@@ -966,7 +966,9 @@ def build_key_value(i, k, v):
             # We bind the new_symint to graph input.
             set_example_value(sym_node_proxy.node, new_symint)
-            self.tx.output.bound_symbols.add(new_symint.node.expr)
+            self.tx.output.root_tracer.bound_symbols[
+                new_symint.node.expr
+            ] = sym_node_proxy
```
can we assert somehow that these are "root symints" or whatever they're called

zou3519 commented on torch/_dynamo/variables/builder.py:
```diff
@@ -1861,6 +1861,7 @@ def update_frame_state(value):
             source=self.get_source(),
         )
 
+        self.tx.output.root_tracer.bound_symbols[wrapped_value.node.expr] = proxy
```
ditto here

ydwu4 committed:
Update on "[hop free symbols][refactor] make bound_symbols a dictionary"


Code refactoring only. Change all self.tx.output.bound_symbols to self.tx.output.root_tracer.bound_symbols.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop free symbols][refactor] make bound_symbols a dictionary"


Code refactoring only. Change all self.tx.output.bound_symbols to self.tx.output.root_tracer.bound_symbols.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop free symbols][refactor] make bound_symbols a dictionary"


Code refactoring only. Change all self.tx.output.bound_symbols to self.tx.output.root_tracer.bound_symbols.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop free symbols][refactor] make bound_symbols a dictionary"


Code refactoring only. Change all self.tx.output.bound_symbols to self.tx.output.root_tracer.bound_symbols.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]
